mmdet/structures/bbox/bbox_mode.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Callable, Optional, Tuple, Type, Union
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from mmdet.structures.mask.structures import BitmapMasks, PolygonMasks
+from .base_bbox import BaseBoxes
+
+BoxType = Union[np.ndarray, Tensor, BaseBoxes]
+MaskType = Union[BitmapMasks, PolygonMasks]
+
+bbox_modes: dict = {}
+_bbox_mode_to_name: dict = {}
+bbox_mode_converters: dict = {}
+
+
+def _register_bbox_mode(name: str,
+                        bbox_mode: Type,
+                        force: bool = False) -> None:
+    """Register a box mode.
+
+    Args:
+        name (str): The name of box mode.
+        bbox_mode (type): Box mode class to be registered.
+        force (bool): Whether to override an existing class with the same
+            name. Defaults to False.
+    """
+    assert issubclass(bbox_mode, BaseBoxes)
+    name = name.lower()
+
+    if not force and (name in bbox_modes or bbox_mode in _bbox_mode_to_name):
+        raise KeyError('name or bbox_mode has been register')
+    elif name in bbox_modes:
+        _bbox_mode = bbox_modes.pop(name)
+        _bbox_mode_to_name.pop(_bbox_mode)
+    elif bbox_mode in _bbox_mode_to_name:
+        _name = _bbox_mode_to_name.pop(bbox_mode)
+        bbox_modes.pop(_name)
+
+    bbox_modes[name] = bbox_mode
+    _bbox_mode_to_name[bbox_mode] = name
+
+
+def register_bbox_mode(name: str,
```

Comment:
for bbox mode, we usually refer to `xywh` and `x1y1x2y2`.  We could simply use register_box.